bitcoin/src/pow.rs

```diff
@@ -421,32 +421,26 @@ impl U256 {
 
     /// Wrapping multiplication by `u64`.
     ///
-    /// Returns a tuple of the addition along with a boolean indicating whether an arithmetic
-    /// overflow would occur. If an overflow would have occurred then the wrapped value is returned.
+    /// # Returns
+    ///
+    /// The multiplication result along with a boolean indicating whether an arithmetic overflow
+    /// occurred. If an overflow occurred then the wrapped value is returned.
     fn mul_u64(self, rhs: u64) -> (U256, bool) {
-        // Multiply 64 bit parts of `mul` by `rhs`.
-        fn mul_parts(mul: u128, rhs: u64) -> (u128, u128) {
-            let upper = (rhs as u128) * (mul >> 64);
-            let lower = (rhs as u128) * (mul & 0xFFFF_FFFF_FFFF_FFFF);
-            (upper, lower)
-        }
+        let mut carry: u128 = 0;
+        let mut split_le = [self.1 as u64, (self.1 >> 64) as u64, self.0 as u64, (self.0 >> 64) as u64];
```

Comment:
This is missing a comment about intentional truncation but maybe it's obvious enough from variable name.

Reply:
Yeah I tried using `buf` as the name and adding a comment but I think its clean like it is.

Reply:
I find `split_le` more clear than `buf`.